crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -47,13 +49,29 @@ module sui::coin {
         icon_url: Option<Url>
     }
 
+    /// Similar to CoinMetadata, but created only for regulated coins that use the DenyList.
+    /// This object is always immutable.
+    struct RegulatedCoinMetadata<phantom T> has key {
+        id: UID,
+        /// The ID of the coin's CoinMetadata object.
+        coin_metadata_object: ID,
+        /// The ID of the coin's DenyCap object.
+        deny_cap_object: ID,
+    }
+
     /// Capability allowing the bearer to mint and burn
     /// coins of type `T`. Transferable
     struct TreasuryCap<phantom T> has key, store {
         id: UID,
         total_supply: Supply<T>
     }
 
+    /// Capability allowing the bearer to freeze addresses, preventing those addresses from
+    /// interacting with the coin as an input to a transaction.
+    struct DenyCap<phantom T> has key, store {
```

Comment:
Should we consider moving this object to the denylist module? It feels odd that it manages an object defined in a different module.

Reply:
The `T` here is specific to the `Coin`'s `T`. The `DenyList` itself pushes off authentication to each use case (which isn't necessarily limited to coin)